sensor/kubernetes/certrefresh/securedcluster_tls_issuer.go

```diff
@@ -234,25 +231,11 @@ func (i *tlsIssuerImpl) ProcessMessage(msg *central.MsgToSensor) error {
 }
 
 func (i *tlsIssuerImpl) dispatch(response *Response) {
-	if !i.requestOngoing.Load() {
-		log.Warnf("Received a response for request ID %q, but there is no ongoing RequestCertificates call.", response.RequestId)
-		return
-	}
-	ongoingRequestID := concurrency.WithLock1(&i.ongoingRequestIDMutex, func() string {
-		return i.ongoingRequestID
-	})
-	if ongoingRequestID != response.RequestId {
-		log.Warnf("Request ID %q does not match ongoing request ID %q.",
-			response.RequestId, ongoingRequestID)
-		return
-	}
-
-	i.responseFromCentral.Store(response)
-	i.responseReceived.Signal()
+	i.responseFromCentral <- response
```

Comment:
The `dispatch` goroutine might leak on stop. Not a big problem in production since sensor only stops if the pod is being stopped but for integration test it might be a problem.

Reply:
Should be fixed now.